src/drivers/imx/sdma.c

```diff
@@ -508,8 +519,12 @@ static int sdma_stop(struct dma_chan_data *channel)
 {
 	struct sdma_chan *pdata = dma_chan_get_data(channel);
 
-	if (channel->status == COMP_STATE_INIT)
-		return -EINVAL;
+	/* do not try to stop multiple times */
+	if (channel->status != COMP_STATE_ACTIVE &&
```

Comment:
Should this be || instead of &&?

Reply:
I think  && is correct. We arrive at sdma_stop from either COMP_STATE_ACTIVE or PAUSED. If there is any other state then this too, this means that stop was already called so we just exit.

Same thing happens for example in src/drivers/dw/dma.c

Reply:
Ahh yes. I misread the code there. sorry for the noise!